lib/tools/system-calls.js

```diff
@@ -419,15 +420,24 @@ systemCallMethods.checkAvdExist = async function (avdName) {
   } catch (e) {
     let unknownOptionError = new RegExp("unknown option: -list-avds", "i").test(e.stderr);
     if (!unknownOptionError) {
-      log.errorAndThrow(`Error executing checkAvdExist. Original error: '${e.message}'; ` +
-                        `Stderr: '${(e.stderr || '').trim()}'; Code: '${e.code}'`);
+      log.errorAndThrow(`Error executing checkAvdExist. Original error: '${e.message}'; Stderr: '${(e.stderr || '').trim()}'; Code: '${e.code}'`);
+    }
 
+    let version = await getSdkToolsVersion();
+    let versionNumber = version.substr(0, version.indexOf("."));
+    if (versionNumber < 25) {
+      // If SDK > 25 use avdmanager instead of android command
+      cmd = await this.getSdkBinaryPath('avdmanager');
+      args = ['list', 'avd', '-c'];
```

Comment:
exec could be extracted from if construction, since it's common:

```javascript
const managerBinaryName = versionNumber >= 25 ? 'avdmanager' ? 'android';
const cmd = await this.getSdkBinaryPath(managerBinaryName);
const result = await exec(cmd, ['list', 'avd', '-c']);
```